packages/ui-components/src/chat/message.tsx

```diff
@@ -127,6 +151,39 @@ export function getFormattedTimestamp(date: string): string {
     return format(parseISO(date), 'HH:mm dd/MM/yyyy');
 }
 
+/**
+ * A function to assign a color to user token depending on their name
+ */
+function selectColor(name: string, colors: string[]): string {
+    // Convert the name to lowercase for consistency
+    const lowerCaseName = name.toLowerCase();
+
+    // Calculate the sum of ASCII values of the characters in the name
+    let asciiSum = 0;
+    for (const char of lowerCaseName) {
+        asciiSum += char.charCodeAt(0);
+    }
+
+    // Use the remainder to select a color
+    const colorIndex = asciiSum % colors.length;
+    return colors[colorIndex];
+}
+
+/**
+ * A function to get the user's initials
+ */
+function getInitials(name: string): string {
+    const parts = name.trim().split(/\s+/);
+    let initials = parts[0][0];
+
+    // If there's a second part, add its first character, so we only get two letter initials
+    if (parts.length > 1) {
+        initials += parts[1][0];
+    }
```

Comment:
What about names with more than one space in it?

Reply:
We just get the first two letters to show as initials, just so it fits well within the bubble. We ignore any further initials after that. Just think that as you add the letters would need to be smaller which seems a bit weird, but happy for suggestions :)

Reply:
I was just thinking that perhaps we want the first and last initial, in case there are middle names or some prefixes which exist in some languages. But either way there will be edge cases so I think it's fine

Reply:
Happy with first and last, will change to do that